### PR TITLE
Scaffold IsFixedLength() instead of IsFixedLength(true)

### PR DESCRIPTION
**Pull request by KaloyanIT**
#23473 

- Change annotation code generation for `IsFixedLength()`

### CODE REVIEW

File: test/EFCore.Design.Tests/Scaffolding/Internal/CSharpDbContextGeneratorTest.cs [hunk 1]
```diff
@@ -816,6 +816,79 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+                        "Employee",
+                        x =>
+                        {
+                            x.Property<int>("Id");
+                            x.Property<string>("Name").IsRequired().HasMaxLength(5).IsFixedLength();
+                        }),
+                new ModelCodeGenerationOptions { UseDataAnnotations = false },
+                code =>
+                {
+                    AssertFileContents(
```
Comment (roji) [inline test/EFCore.Design.Tests/Scaffolding/Internal/CSharpDbContextGeneratorTest.cs:834]: nit: probably enough to assert that `.IsFixedLength()` is contained in the scaffolded code (e.g. ComputedColumnSql_works_unspecified), to keep things short and less brittle.

File: test/EFCore.Design.Tests/Scaffolding/Internal/CSharpDbContextGeneratorTest.cs [hunk 2]
```diff
@@ -816,6 +816,79 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+        }
+
+        partial void OnModelCreatingPartial(ModelBuilder modelBuilder);
+    }
+}
+",
+                        code.ContextFile);
+                },
+                model =>
+                    Assert.Equal(true, model.FindEntityType("TestNamespace.Employee").GetProperty("Name").GetIsFixedLength()));
```
Comment (roji) [inline test/EFCore.Design.Tests/Scaffolding/Internal/CSharpDbContextGeneratorTest.cs:889]:
```suggestion
                    Assert.True(model.FindEntityType("TestNamespace.Employee").GetProperty("Name").IsFixedLength()));
```

File: src/EFCore.Relational/Metadata/Internal/RelationalPropertyExtensions.cs [hunk 3]
```diff
@@ -25,5 +25,15 @@ public static class RelationalPropertyExtensions
...
             => (string?)property[RelationalAnnotationNames.ColumnType];
+
+        /// <summary>
+        ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+        ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+        ///     any release. You should only use it directly in your code with extreme caution and knowing that
+        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+        /// </summary>
+        [DebuggerStepThrough]
+        public static bool? GetIsFixedLength([NotNull] this IProperty property)
```
Comment (roji) [inline src/EFCore.Relational/Metadata/Internal/RelationalPropertyExtensions.cs:36]: This shouldn't be needed as we already have IsFixedLength

**Review by roji — COMMENTED** [review]
Thanks, looks good - see some minor corrections.

File: src/EFCore.Relational/Metadata/Internal/RelationalPropertyExtensions.cs (hunk 3 above)
Comment (KaloyanIT) [reply src/EFCore.Relational/Metadata/Internal/RelationalPropertyExtensions.cs]: Thank you for the review

File: test/EFCore.Design.Tests/Scaffolding/Internal/CSharpDbContextGeneratorTest.cs [hunk 4]
```diff
@@ -816,6 +816,79 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                     Assert.Equal("date", model.FindEntityType("TestNamespace.Employee").GetProperty("HireDate").GetConfiguredColumnType()));
         }
 
+        [ConditionalFact]
+        public void Is_fixed_legth_annotation_should_be_scafollded_without_optional_parameter()
```
[the remark this answers is not in the document]
Comment (roji) [reply test/EFCore.Design.Tests/Scaffolding/Internal/CSharpDbContextGeneratorTest.cs:820]: Typos are still here

**Review by roji — APPROVED** [review]
Thanks!

**Comment by Saibamen** [comment]
Can this be shipped with 5.0.2?

**Comment by roji** [comment]
@Saibamen as this is purely a cosmetic issue rather than a bug, there's little chance this could be considered for a 5.0 patch...

**Comment by Saibamen** [comment]
So there is nothing to wait for. This is just cosmetic issue, so there is no breaking changes

**Comment by roji** [comment]
@Saibamen you probably want to take a look at [patching policy](https://docs.microsoft.com/en-us/ef/core/what-is-new/release-planning#patch-releases), specifically about the part where patches are about fixing bugs. Any change implies risk, so we don't merge cosmetics to patch releases.

**Pull request merged**